test/extended/baremetal/helper.go

```diff
@@ -0,0 +1,214 @@
+	return fmt.Sprintf("extraworker-%d-secret", index)
+}
+
+func (b *BaremetalTestHelper) extraWorkerKey(index int) string {
+	return fmt.Sprintf("extraworker-%d-bmh", index)
+}
+
+// CanDeployExtraWorkers checks if current platform contains
+// the necessary data to deploy additional workers
+func (b *BaremetalTestHelper) CanDeployExtraWorkers() bool {
+	_, err := b.extraWorkersRetrieveData()
+	return err == nil
+}
+
+// getExtraWorkerSecretData gets and decodes the secret associated for the
+// specified extra worker. If not found, the test will fail
+func (b *BaremetalTestHelper) getExtraWorkerSecretData(index int) *v1.Secret {
+	ew, err := b.extraWorkersRetrieveData()
+	o.Expect(err).ToNot(o.HaveOccurred())
+
+	yamlData, ok := ew.Data[b.extraWorkersSecretKey(index)]
+	o.Expect(ok).To(o.BeTrue(), fmt.Sprintf("unable to find data for extra worker %d", index))
+
+	var secret v1.Secret
+	err = yaml.Unmarshal(yamlData, &secret)
+	o.Expect(err).ToNot(o.HaveOccurred())
+	return &secret
+}
+
+// getExtraWorkerData gets and decodes the baremetal host associated for the
+// specified extra worker. If not found, the test will fail
+func (b *BaremetalTestHelper) getExtraWorkerData(index int) *unstructured.Unstructured {
+	ew, err := b.extraWorkersRetrieveData()
+	o.Expect(err).ToNot(o.HaveOccurred())
+
+	yamlData, ok := ew.Data[b.extraWorkerKey(index)]
+	o.Expect(ok).To(o.BeTrue(), fmt.Sprintf("unable to find data for extra worker %d", index))
+
+	jsonData, err := yaml.YAMLToJSON(yamlData)
+	o.Expect(err).ToNot(o.HaveOccurred())
+
+	var host unstructured.Unstructured
+	err = host.UnmarshalJSON(jsonData)
+	o.Expect(err).ToNot(o.HaveOccurred())
+
+	return &host
+}
+
+// GetExtraWorkerData gets baremetal host and secret for the specified extra worker.
+// If not found, the test will fail
+func (b *BaremetalTestHelper) GetExtraWorkerData(index int) (*unstructured.Unstructured, *v1.Secret) {
+	return b.getExtraWorkerData(index), b.getExtraWorkerSecretData(index)
+}
+
+// DeployExtraWorker is an utility function that creates the specified worker
+// and waits until it reaches the Available state
+func (b *BaremetalTestHelper) DeployExtraWorker(index int) (*unstructured.Unstructured, *v1.Secret) {
+	hostData, secretData := b.GetExtraWorkerData(index)
+	host, secret := b.CreateExtraWorker(hostData, secretData)
+	b.WaitForProvisioningState(host, "available")
+	return host, secret
+}
+
+// CreateExtraWorker creates a new BaremetalHost (and associated Secret).
+// If successfull, returns the newly created host and secret resources
+func (b *BaremetalTestHelper) CreateExtraWorker(host *unstructured.Unstructured, secret *v1.Secret) (*unstructured.Unstructured, *v1.Secret) {
+
+	secret, err := b.clientSet.CoreV1().Secrets("openshift-machine-api").Create(context.Background(), secret, metav1.CreateOptions{})
+	o.Expect(err).ToNot(o.HaveOccurred())
+
+	host, err = b.bmcClient.Create(context.Background(), host, metav1.CreateOptions{})
```

Comment:
Wondering if its intentional to use context.Background() here and context.TODO() in other places (e.g. line 149)?